testutil/integration/router.go

```diff
@@ -80,14 +82,27 @@ func NewIntegrationApp(
 		return moduleManager.EndBlock(sdkCtx)
 	})
 
-	msgRouter := baseapp.NewMsgServiceRouter()
 	msgRouter.SetInterfaceRegistry(interfaceRegistry)
 	bApp.SetMsgServiceRouter(msgRouter)
+	grpcRouter.SetInterfaceRegistry(interfaceRegistry)
+	bApp.SetGRPCQueryRouter(grpcRouter)
 
 	if keys[consensusparamtypes.StoreKey] != nil {
 		// set baseApp param store
-		consensusParamsKeeper := consensusparamkeeper.NewKeeper(appCodec, runtime.NewEnvironment(runtime.NewKVStoreService(keys[consensusparamtypes.StoreKey]), log.NewNopLogger(), runtime.EnvWithRouterService(baseapp.NewGRPCQueryRouter(), msgRouter)), authtypes.NewModuleAddress("gov").String())
+		consensusParamsKeeper := consensusparamkeeper.NewKeeper(appCodec, runtime.NewEnvironment(runtime.NewKVStoreService(keys[consensusparamtypes.StoreKey]), log.NewNopLogger(), runtime.EnvWithRouterService(grpcRouter, msgRouter)), authtypes.NewModuleAddress("gov").String())
 		bApp.SetParamStore(consensusParamsKeeper.ParamsStore)
+		consensusparamtypes.RegisterQueryServer(grpcRouter, consensusParamsKeeper)
+
+		_, err := consensusParamsKeeper.SetParams(sdkCtx, &consensusparamtypes.ConsensusMsgParams{
```

Comment:
I don't follow why this is happening here, don't consensus params get set during InitGenesis?

Reply:
initgenesis doesnt run in integration tests only e2e tests.